### PR TITLE
Add user management for alpine-based images.

### PR DESCRIPTION
**Pull request by marat-biriushev**
# Task
[DOPS-2586](https://soramitsu.atlassian.net/browse/DOPS-2586): Add user management for alpine-based images in sora-gradle-plugin.

# Changes
Added condition for alpine based OS.
# Author
Signed-off-by: Marat Biryushev [vfhfnvvv@gmail.com](mailto:vfhfnvvv@gmail.com)

[DOPS-2586]: https://soramitsu.atlassian.net/browse/DOPS-2586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ plugins {
 
 group 'jp.co.soramitsu'
 def pluginId = "${group}.sora-plugin"
-version System.getenv('TAG_NAME') ?: "0.5.3"
+version System.getenv('TAG_NAME') ?: "0.5.4-rc1"
```
Comment (wpi86) [inline build.gradle:16]: don't forget

**Review by wpi86 — APPROVED** [review]

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy [hunk 2]
```diff
@@ -215,7 +215,16 @@ class DockerPlugin implements Plugin<Project> {
             t.entryPoint "/tini", "--"
 
             // add user
-            t.runCommand "groupadd -r appuser && useradd -r -g appuser appuser"
+            //def command = "cat /etc/os-release"
```
Comment (Mingela) [inline src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:218]: remove these comments?

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy [hunk 3]
```diff
@@ -215,7 +215,16 @@ class DockerPlugin implements Plugin<Project> {
...
+            //def content = ["sh", "-c", command].execute().text
+            def command = "docker run -t --rm ${dockerConfig.baseImage} cat /etc/os-release"
+            def content = ["sh", "-c", command].execute().text
+            println content
+            if ( content.contains("alpine") ) {
```
Comment (Mingela) [inline src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:223]:
```suggestion
            if (content.contains("alpine")) {
```

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy [hunk 4]
```diff
@@ -215,7 +215,16 @@ class DockerPlugin implements Plugin<Project> {
...
+                t.runCommand "addgroup -S appuser && adduser -S -G appuser appuser"
+            }  else {
+                t.runCommand "groupadd -r appuser && useradd -r -g appuser appuser"
+            }
```
Comment (Mingela) [inline src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:227]:
```suggestion
            def groupCommand
            if (content.contains("alpine")) {
                groupCommand = "addgroup -S appuser && adduser -S -G appuser appuser"
            }  else {
                groupCommand = "groupadd -r appuser && useradd -r -g appuser appuser"
            }        
             t.runCommand groupCommand    
```

File: build.gradle [hunk 5]
```diff
@@ -13,7 +13,7 @@ plugins {
 
 group 'jp.co.soramitsu'
 def pluginId = "${group}.sora-plugin"
-version System.getenv('TAG_NAME') ?: "0.5.3"
+version System.getenv('TAG_NAME') ?: "0.5.4-rc3"
```
Comment (Mingela) [inline build.gradle:16]: we shouldn't create new tags without real releases, please consider removing all unnecessary tags and replace that with just `0.5.4`, we can make a release right after merge

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 2 above)
Comment (marat-biriushev) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy]: Done.

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 4 above)
Comment (marat-biriushev) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy]: Done.

File: build.gradle (hunk 5 above)
Comment (marat-biriushev) [reply build.gradle]: Done.

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy [hunk 6]
```diff
@@ -215,7 +215,15 @@ class DockerPlugin implements Plugin<Project> {
...
-            t.runCommand "groupadd -r appuser && useradd -r -g appuser appuser"
+            def command = "docker run -t --rm ${dockerConfig.baseImage} cat /etc/os-release"
+            def content = ["sh", "-c", command].execute().text
+
+            if (content.contains("alpine")) {
+                def groupCommand = "addgroup -S appuser && adduser -S -G appuser appuser"
+            }  else {
+                def groupCommand = "groupadd -r appuser && useradd -r -g appuser appuser"
+            }
+            t.runCommand groupCommand
```
Comment (Mingela) [inline src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:226]: shouldn't that be like the following?
```suggestion
            def groupCommand = ""
            if (content.contains("alpine")) {
                groupCommand = "addgroup -S appuser && adduser -S -G appuser appuser"
            }  else {
                groupCommand = "groupadd -r appuser && useradd -r -g appuser appuser"
            }
            t.runCommand groupCommand             
```

**Review by Mingela — APPROVED** [review]

**Pull request merged**